core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -159,11 +159,30 @@
    *
    * <p>If this method is not called, the builder will use {@link PickFirstBalancerFactory}
    * for the channel.
+   *
+   * <p>This overrides what was set by {@link #loadBalancerFactory(LoadBalancer2.Factory)}.
    */
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1771")
   public abstract T loadBalancerFactory(LoadBalancer.Factory loadBalancerFactory);
 
   /**
+   * Provides a custom {@link LoadBalancer2.Factory} for the channel.
+   *
+   * <p>If this method is not called, the builder will use {@link PickFirstBalancerFactory}
+   * for the channel.
+   *
+   * <p>This overrides what was set by {@link #loadBalancerFactory(LoadBalancer.Factory)}.
+   *
+   * <p><strong>IMPORTANT: </strong>Calling this will make the channel to run the LBv2 code path,
+   * which is new and not as well-exercised as the current one.  You should use it only if you want
+   * to try out LBv2, or you are migrating your LoadBalancer implementation to LBv2.  See
+   * <a href="https://github.com/grpc/grpc-java/issues/2656" target="_blank">#2656</a> for more
+   * information.
+   */
```

Comment:
No, it won't. Next version will still have two signatures, while `LoadBalancer2` will just be an alias of `LoadBalancer`.

At 1.1, we expect users of stock LoadBalancers to keep using the first signature, and users of custom LoadBalancers to migrate to the second signature. Therefore, neither of them are deprecated.

At 1.2, the one with `LoadBalancer2` will be deprecated.